JASP-Engine/JASP/R/descriptives.R

```diff
@@ -1253,4 +1275,45 @@ Descriptives <- function(jaspResults, dataset, options) {
   return(descriptivesQQPlot)
 }
 
+.descriptivesPieChart <- function(dataset, options, variable) {
+  if (options$splitby != "" ) {
+    # return a collection
+    split <- names(dataset)
+
+    plotResult <- createJaspContainer(title=variable)
+    plotResult$dependOn(options="splitby", optionContainsValue=list(variables=variable))
+
+    for (l in split) {
+      plotResult[[l]] <- .descriptivesPieChart_SubFunc(column=dataset[[l]][[.v(variable)]], variable=variable, width=options$plotWidth, height=options$plotHeight, displayDensity = options$distPlotDensity, title = l)
+      plotResult[[l]]$dependOn(optionsFromObject=plotResult)
+    }
+
+    return(plotResult)
+  } else {
+    column <- dataset[[.v(variable)]]
+    aPlot <- .descriptivesPieChart_SubFunc(column=column[!is.na(column)], variable=variable, width=options$plotWidth, height=options$plotHeight, displayDensity = options$distPlotDensity, title = variable)
+    aPlot$dependOn(options="splitby", optionContainsValue=list(variables=variable))
+
+    return(aPlot)
+  }
+}
+
+.descriptivesPieChart_SubFunc <- function(column, variable, width, height, displayDensity, title) {
+  plotObj <- createJaspPlot(title=title, width=width, height=height)
+
+  if (any(is.infinite(column))) {
+    plotObj$setError("Plotting is not possible: Variable contains infinity")
+    plotObj$plotObject    <- JASPgraphs::themeJasp(JASPgraphs::drawAxis(xName = variable, xBreaks = 1:5, yBreaks = 1:5))
```

Comment:
```suggestion
```
If you set an error, there should be no need to fill the plotObject.

Reply:
noted